api/scpca_portal/models/library.py

```diff
@@ -104,3 +109,45 @@ def get_formats_from_file_paths(cls, file_paths: List[Path]) -> List[str]:
         if any(path for path in file_paths if common.ANNDATA_EXT == path.suffix):
             formats.append(Library.FileFormats.ANN_DATA)
         return formats
+
+    @classmethod
+    def get_project_libraries_from_download_config(
+        cls, project, download_configuration: Dict
+    ) -> List[Self]:
+        if download_configuration not in common.GENERATED_PROJECT_DOWNLOAD_CONFIGURATIONS:
+            raise ValueError("Invalid download configuration passed. Unable to retrieve libraries.")
+
+        if download_configuration["includes_merged"]:
+            # If the download config requests merged and there is no merged file in the project,
+            # return an empty list
+            if (not project.includes_merged_sce) and (not project.includes_merged_anndata):
+                return []
+
+        if download_configuration["metadata_only"]:
+            return project.libraries.all()
```

Comment:
I don't think there are any potential pitfalls with having this here but I think it would be better placed at the top since it is the least restrictive check. I can see that you put it here because it is not a "guard"and generally those go first as an escape hatch but I think that because it is checked independently of any other values it's clearer up top (after the first validation guard).